h2o-bindings/bin/custom/python/gen_gbm.py

```diff
@@ -0,0 +1,63 @@
+class_doc = """
+Builds gradient boosted trees on a parsed data set, for regression or classification.
+The default distribution function will guess the model type based on the response column type.
+Otherwise, the response column must be an enum for "bernoulli" or "multinomial", and numeric
+for all other distributions.
+"""
+
```

Comment:
@angela0xdata @hannah-tillman this is how the template for examples will look:
this simple `python/gen_gbm.py` file can define some basic string variables into which you can just add your doc and examples.

For object properties examples, what would you prefer?
1. a flat structure like 
``` 
property_training_frame_examples = ... 
property_validation_frame_examples = ...
```
2. a dictionary:
```
properties_examples = dict(
    training_frame="""...""",
    validation_frame="""...""",
)
```
see below to see how it looks like for real.
The second one may be slightly more readable, but don't forget your commas :)

Reply:
So to provide an example for `balance_classes` in GBM, we can just add to that flat file or dict?
(flat file example)

```
property_balance_classes_examples = """
>>> cars = h2o.import_file("https://s3.amazonaws.com/h2o-public-test-data/smalldata/junit/cars_20mpg.csv")
>>> cars["economy_20mpg"] = cars["economy_20mpg"].asfactor()
>>> predictors = ["displacement","power","weight","acceleration","year"]
>>> response = "economy_20mpg"
>>> train, valid = cars.split_frame(ratios = [.8], seed = 1234)
>>> cars_gbm = H2OGradientBoostingEstimator(balance_classes=True, seed = 1234)
>>> cars_gbm.train(x = predictors,
...                y = response,
...                training_frame = train,
...                validation_frame = valid)
>>> cars_gbm.auc(valid=True)
```

Reply:
@angela0xdata 👍 for (1) or for (2)? :)
I won't support both.
As I'm externalizing more of those customizations, I tend to go towards this dictionary structure:
```
properties = dict(
     training_frame = dict(
          examples = """
>>> statement
>>> another statement
         """,
    ),
    validation_frame = dict(
          examples = """
>>> statement
>>> another statement
         """,
    ),
    some_property_with_many_customizations = dict(
        doc = """ This property also does this and that""",  # doc will be added to getter only
        examples = """cf. above""",  # examples will be added to getter only
        getter = """
        if isinstance(prop, str):
            return converted_to_list
        else:
            return prop
        """,
        setter = """
        blah blah blah
        """
    ),
)
```

it hurts a bit more the eyes here, but looks fine with color syntax. 
Would it be fine for you?

Reply:
Either way would work for me, and I have no preference :) ... I just wanted to make sure I understood correctly.

Reply:
@angela0xdata yep, I didn't see your previous comment, you got it right